com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -186,6 +222,64 @@ public Class visitEndProgramStatement(EndProgramStatementContext ctx) {
     return visitChildren(ctx);
   }
 
+  @Override
+  public Class visitProgramIdParagraph(ProgramIdParagraphContext ctx) {
+    Optional<String> programName = Optional.ofNullable(ctx.programName()).map(RuleContext::getText);
+    if (programName.isPresent()) {
+      String name = programName.get();
+      this.programName = name;
+      outlineTreeBuilder.renameProgram(name, ctx);
+      outlineTreeBuilder.addNodeToParent("PROGRAM-ID " + name, NodeType.PROGRAM_ID, ctx);
+    }
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitConfigurationSection(ConfigurationSectionContext ctx) {
+    outlineTreeBuilder.addNodeToParent("CONFIGURATION SECTION", NodeType.SECTION, ctx);
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitInputOutputSection(InputOutputSectionContext ctx) {
+    outlineTreeBuilder.addNodeToParent("INPUT-OUTPUT SECTION", NodeType.SECTION, ctx);
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitSelectClause(SelectClauseContext ctx) {
+    outlineTreeBuilder.addNodeToParent(ctx.fileName().getText(), NodeType.FILE, ctx);
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitFileSection(FileSectionContext ctx) {
+    outlineTreeBuilder.addNodeToParent("FILE SECTION", NodeType.SECTION, ctx);
+    outlineTreeBuilder.initVariables();
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitLinkageSection(LinkageSectionContext ctx) {
+    outlineTreeBuilder.addNodeToParent("LINKAGE SECTION", NodeType.SECTION, ctx);
+    outlineTreeBuilder.initVariables();
+    return visitChildren(ctx);
+  }
+
+  @Override
+  public Class visitLocalStorageSection(LocalStorageSectionContext ctx) {
+    outlineTreeBuilder.addNodeToParent("LOCAL STORAGE SECTION", NodeType.SECTION, ctx);
```

Comment:
Would be better to rename this method since what is doing at this level is to add a new node. the inner private method will be responsible to put that node at the right level